filesystem/hdfs.go

```diff
@@ -0,0 +1,165 @@
+package filesystem
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"regexp"
+
+	"github.com/colinmarc/hdfs"
+)
+
+// Requirement:
+//   hadoop version:
+//   HDFS version:
+//   REST API version:
+
+type hdfsConfig struct {
+	namenodeAddr string
+	webHdfsAddr  string
+	user         string
+}
+
+type HdfsClient struct {
+	client *hdfs.Client
+	hdfsConfig
+}
+
+func NewHdfsClient(namenodeAddr, webHdfsAddr, user string) (Client, error) {
+	client, err := hdfs.NewForUser(namenodeAddr, user)
+	if err != nil {
+		return nil, err
+	}
+	// client.Remove("/tmp/logs/testing")
+	return &HdfsClient{
+		client: client,
+		hdfsConfig: hdfsConfig{
+			namenodeAddr: namenodeAddr,
+			webHdfsAddr:  webHdfsAddr,
+			user:         user,
+		},
+	}, nil
+}
+
+func (c *HdfsClient) OpenReadCloser(name string) (io.ReadCloser, error) {
+	return c.client.Open(name)
+}
+
+func (c *HdfsClient) OpenWriteCloser(name string) (io.WriteCloser, error) {
+	exist, err := c.Exists(name)
+	if err != nil {
+		return nil, err
+	}
+	if !exist {
+		err := c.client.CreateEmptyFile(name)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &HdfsFile{
+		path:       name,
+		logger:     log.New(os.Stdout, "", log.Lshortfile|log.LstdFlags),
+		hdfsConfig: c.hdfsConfig,
+	}, nil
+}
+
+func (c *HdfsClient) Exists(name string) (bool, error) {
+	_, err := c.client.Stat(name)
+	return existCommon(err)
+}
+
+func (c *HdfsClient) Rename(oldpath, newpath string) error {
+	return c.client.Rename(oldpath, newpath)
+}
+
+func (c *HdfsClient) Glob(dirname, pattern string) ([]string, error) {
+	fileInfoList, err := c.client.ReadDir(dirname)
```

Comment:
It looks like you have specific requirements on path pattern.
(*, ?, etc)
(*, ?) isn't regex. And what does it mean for "etc"?

I mean it's hard because if it's full regex pattern.
And what you wanna do can be implemented on top of this, right?